src/components/AffiliateBanner.js

```diff
@@ -0,0 +1,113 @@
+import React, {useEffect} from 'react';
+import '../css/AffiliateBanner.css';
+import LogoImg from '../images/soulsmile-1024.png';
+import ComputerImg from '../images/computer_affiliate_version.png';
+import Button from 'react-bootstrap/Button';
+
+import GirlfriendImg from '../images/product_girlfriend_img.jpeg';
+import ByHumankindImg from '../images/product_byhumankind_img.jpeg';
+import CicadaImg from '../images/product_cicada_img.jpeg';
+import RocketbookImg from '../images/product_rocketbook_img.jpeg';
+
+var REACT_APP_AIRTABLE_RETAILERS_DOC = process.env.REACT_APP_AIRTABLE_RETAILERS_DOC;
+
+function AffiliateBanner(props) {
+
+    const { collab } = props;
+    //URLSearchParams Attempt
+    // //change above: function AffiliateBanner({ location }, props)
+    // function useQuery() {
+    //     return new URLSearchParams(location.search);
+    // }
+
+    // let query = useQuery();
+    // var affiliate = parseFloat(query.get("affiliate"));
+
+    // if(affiliate == 'girlfriend') {
+    //     collab = 'Girlfriend Collective';
+    // }
+
+    const [adRetailer, setAdRetailer] = React.useState([]);
+    
+    
+    useEffect(() => {
+        fetch(REACT_APP_AIRTABLE_RETAILERS_DOC)
+			.then(res => res.json())
+			.then(res => {
+                const data = res.records;
+                data.sort(function (a, b) {
+                    return ((a["fields"]["Name"] < b["fields"]["Name"]) ? -1 : (a["fields"]["Name"] > b["fields"]["Name"]) ? 1 : 0);
+                });
+
+                //Set product_img
+                var product_img;
+                switch(collab) {
+                    case 'Girlfriend Collective':
+                        product_img = GirlfriendImg;
+                        break;
+                    case 'by Humankind':
+                        product_img = ByHumankindImg;
+                        break;
+                    case 'Cicada':
+                        product_img = CicadaImg;
+                        break;
+                    case 'Rocketbook':
+                        product_img = RocketbookImg;
+                        break;
+                    default:
+                        product_img = null;
+                }
+                
+                var currentAdRetailer = [];
+                for (var j = 0; j < data.length; j++) {
+                    const company = data[j]["fields"]["Name"];
+                    const link = data[j]["fields"]["Link"] + "?fromTab=true";
+
+                    if (company === collab) {
+                        currentAdRetailer = [company, link, product_img];
```

Comment:
I also think this might be clearer/less bug-prone in the future if we have 3 separate variables for the retailerName, retailerLink, and productImage!

Reply:
I tried to use 3 separate variables and got a "href.trim is not a function" error - not immediately clear to me what is causing this when I did a google search. I might be making some sort of mistake with react hooks? I left the original code as is but I also left the new code I tried (commented out)

Reply:
Update: based on later comment, I moved my attempt to another branch so the commented out attempt doesn't get merged into the master branch